llvm/utils/TableGen/DecoderEmitter.cpp

```diff
@@ -488,8 +502,9 @@ class FilterChooser {
   // Links to the FilterChooser above us in the decoding tree.
   const FilterChooser *Parent;
 
-  // Width of instructions
-  unsigned BitWidth;
+  /// Some targets (ARM) specify more encoding bits in Inst that Size allows.
+  /// This field allows us to ignore the extra bits.
+  unsigned MaxFilterWidth;
```

Comment:
I ran into this, so I am proposing that for fixed width insts, change InstructionEncoding::BitWidth to be Size * 8 and not the size of the `Inst` bits (with some checks). Working on a PR for that.

Reply:
I really woudn't want to hardcode 8 inside InstructionEncoding. The value is wrong on my downstream arch. I would prefer it be hardcoded somewhere else, where we have access to CodeGenTarget, if possible.